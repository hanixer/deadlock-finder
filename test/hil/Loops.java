package showcase;

public class Loops {
    public static void func1(int a, int b) {
        int i = 0;
        while (i < 5) {
            a++;
            i++;
        }

        for (int j = 0, kk = 123; j < 4; j++) {
            b--;
        }
    }
}
