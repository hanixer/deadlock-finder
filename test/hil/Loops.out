func showcase.Loops.func1(a: int, b: int): void {
  var i: int = 0
  while {
    var t~1: boolean = i<5
  } (t~1) {
    a = a+1
    i = i+1
  }
  {
    var j: int = 0
    var kk: int = 123
    while {
      var t~2: boolean = j<4
    } (t~2) {
      b = b-1
      j = j+1
    }
  }
}